app/status.go

```diff
@@ -236,10 +236,21 @@ func IsUserAway(lastActivityAt int64) bool {
 	return model.GetMillis()-lastActivityAt >= *utils.Cfg.TeamSettings.UserStatusAwayTimeout*1000
 }
 
-func DoesStatusAllowPushNotification(user *model.User, status *model.Status, channelId string) bool {
+func DoesStatusAllowPushNotification(user *model.User, channelNotifyProps model.StringMap, wasMentioned bool, status *model.Status, channelId string) bool {
```

Comment:
Maybe rename to `ShouldSendPushNotification`? It's checking both status and props. You could even split those two responsibilities into different functions.